src/main/java/build/buildfarm/common/distributed/DistributedHashMap.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm.common.distributed;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @class DistributedHashMap
+ * @brief A distributed hashmap.
+ * @details A distributed hashmap is an implementation of a hashmap data structure which can be
+ *     shared across a distributed system. For example, implementations of this interface might
+ *     choose to use redis, hazelcast, postgres, etc.
+ */
+public interface DistributedHashMap<DistributedClient> {
```

Comment:
shouldn't this extend the DistributedMap interface? Also the Hash seems like an implementation detail that doesn't mandate a separate interface.

Reply:
That was a good idea.  There was more overlap than I thought.  Now there is only `DistributedMap` and we can create `RedisMap`, `RedisHashMap` from it.